src/components/patterns/headingList.tsx

```diff
@@ -161,6 +163,11 @@ const HeadingList = () => {
   useEventListener("scroll", debouncedHandleScroll)
   useEventListener("hashchange", handleAnchor)
 
+  // Don't render anything until headings are loaded
+  if (!isLoaded) {
+    return null
```

Comment:
I do wonder if it would be worth while to include a skeleton loader for UX purposes, but as-is should fix the SiteImprove issue 🤔

Reply:
It's so fast, it's not worth the trouble. If it were fetching data from the server or something something very time consuming, it might be useful.